### PR TITLE
GH-1504 Create post office manager plugin

### LINKED ISSUE #1504: Create post office manager plugin
After installing a new local RPM based installation of IML I added two MDS nodes that had been configured in monitored mode and noticed that the manager was receiving disconnects every few seconds. Looking into the logs, I discovered the following error message:

```
session create request for postoffice failed: Reqwest(reqwest::Error { kind: Status(502), url: "https://adm.local/agent2/message/" })
Jan 21 16:07:16 mds1.local iml-agent-daemon[8147]: Jan 21 16:07:16.529  INFO iml_agent::poller: session create request for ostpool failed: Reqwest(reqwest::Error { kind: Status(502), url: "https://adm.local/agent2/message/" })
Jan 21 16:07:16 mds1.local iml-agent-daemon[8147]: Jan 21 16:07:16.530  INFO iml_agent::poller: session create request for stratagem failed: Reqwest(reqwest::Error { kind: Status(502), url: "https://adm.local/agent2/message/" })
Jan 21 16:07:16 mds1.local iml-agent-daemon[8147]: Jan 21 16:07:16.530  INFO iml_agent::poller: session create request for action_runner failed: Reqwest(reqwest::Error { kind: Status(502), url: "https://adm.local/agent2/message/" })
```

After discussing, it appears that we need to add a post office manager plugin so that the session is created appropriately.

### PR DESCRIPTION
**Pull request by johnsonw**
Fixes #1504.

After installing a new local RPM based installation of IML I added two MDS nodes that had been configured in monitored mode and noticed that the manager was receiving disconnects every few seconds. Looking into the logs, I discovered the following error message:

```
session create request for postoffice failed: Reqwest(reqwest::Error { kind: Status(502), url: "https://adm.local/agent2/message/" })
Jan 21 16:07:16 mds1.local iml-agent-daemon[8147]: Jan 21 16:07:16.529  INFO iml_agent::poller: session create request for ostpool failed: Reqwest(reqwest::Error { kind: Status(502), url: "https://adm.local/agent2/message/" })
Jan 21 16:07:16 mds1.local iml-agent-daemon[8147]: Jan 21 16:07:16.530  INFO iml_agent::poller: session create request for stratagem failed: Reqwest(reqwest::Error { kind: Status(502), url: "https://adm.local/agent2/message/" })
Jan 21 16:07:16 mds1.local iml-agent-daemon[8147]: Jan 21 16:07:16.530  INFO iml_agent::poller: session create request for action_runner failed: Reqwest(reqwest::Error { kind: Status(502), url: "https://adm.local/agent2/message/" })
```

After discussing, it appears that we need to add a post office manager plugin so that the session is created appropriately.

Signed-off-by: johnsonw <wjohnson@whamcloud.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1505)

### CODE REVIEW

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -266,6 +266,17 @@ services:
       - "manager-config:/var/lib/chroma"
     environment:
       - RUST_LOG=info
+  iml-stratagem:
```
Comment (jgrund) [inline docker/docker-compose.yml]:
```suggestion
  iml-postoffice:
```

**Comment by johnsonw** [comment]
I've marked this ready for review but am currently testing. Will post results when finished.

**Comment by johnsonw** [comment]
Maybe that's what I was missing, i'm getting:
```
Processing files: rust-iml-postoffice-0.1.2-1.1579633693.el7.x86_64
error: File not found: /build/_topdir/BUILDROOT/rust-iml-0.1.2-1.1579633693.el7.x86_64/usr/bin/iml-postoffice
error: File not found: /build/_topdir/BUILDROOT/rust-iml-0.1.2-1.1579633693.el7.x86_64/usr/lib/systemd/system/iml-postoffice.service


RPM build errors:
    File not found: /build/_topdir/BUILDROOT/rust-iml-0.1.2-1.1579633693.el7.x86_64/usr/bin/iml-postoffice
    File not found: /build/_topdir/BUILDROOT/rust-iml-0.1.2-1.1579633693.el7.x86_64/usr/lib/systemd/system/iml-postoffice.service
Traceback (most recent call last):
  File "/usr/bin/create_build", line 82, in <module>
    cwd=workspace,
  File "/usr/lib64/python2.7/subprocess.py", line 542, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['rpmbuild', '--rebuild', '/tmp/rust-iml-0.1.2-1.1579633693.el7.src.rpm', '-D', '%_topdir /build/_topdir']' returned non-zero exit status 1
```

**Comment by utopiabound** [comment]
Should there be a service that handles the opposite end of "empty" sessions?  the postoffice and stratagem services don't do anything, can they do that together?

**Comment by jgrund** [comment]
> Should there be a service that handles the opposite end of "empty" sessions? the postoffice and stratagem services don't do anything, can they do that together?

I've thought about that; really I think postoffice could probably be a standalone service since it doesn't use any of the daemon plugin functionality from what I've seen. 

However, since this is really just a stub at the moment and is blocking master I think we should land this and revisit that.

**Comment by johnsonw** [comment]
Checking on the agent logs the postoffice session was created successfully:
```
Jan 22 14:22:06 mds1.local iml-agent-daemon[1303]: Jan 22 14:22:06.861  INFO iml_agent::http_comms::session: Created new session PluginName("action_runner")/Id("06ccbf68-4a1f-4ad7-9d1f-d67f83adc836")
Jan 22 14:22:06 mds1.local iml-agent-daemon[1303]: Jan 22 14:22:06.989  INFO iml_agent::http_comms::session: Created new session PluginName("stratagem")/Id("83866580-68d9-4461-9ce3-be1b71d9783a")
Jan 22 14:22:06 mds1.local iml-agent-daemon[1303]: Jan 22 14:22:06.990  INFO iml_agent::http_comms::session: Created new session PluginName("postoffice")/Id("54807c17-2d91-4d2e-b534-336d0e3b0ce4")
```

**Comment by johnsonw** [comment]
Client connected and charts working:
image

**Comment by johnsonw** [comment]
Just following up with postoffice and using docker. 

rust-iml-agent logs after adding server:
```
[root@mds1 ~]# journalctl -fu rust-iml-agent
-- Logs begin at Wed 2020-01-22 20:24:31 UTC. --
Jan 22 20:25:10 mds1.local iml-agent-daemon[1668]: Jan 22 20:25:10.335  INFO iml_agent::action_plugins::action_plugin: ostpool_destroy
Jan 22 20:25:10 mds1.local iml-agent-daemon[1668]: Jan 22 20:25:10.335  INFO iml_agent::action_plugins::action_plugin: package_version
Jan 22 20:25:10 mds1.local iml-agent-daemon[1668]: Jan 22 20:25:10.335  INFO iml_agent::action_plugins::action_plugin: stream_fidlists_stratagem
Jan 22 20:25:10 mds1.local iml-agent-daemon[1668]: Jan 22 20:25:10.335  INFO iml_agent::action_plugins::action_plugin: start_unit
Jan 22 20:25:10 mds1.local iml-agent-daemon[1668]: Jan 22 20:25:10.335  INFO iml_agent::action_plugins::action_plugin: enable_unit
Jan 22 20:25:10 mds1.local iml-agent-daemon[1668]: Jan 22 20:25:10.335  INFO iml_agent::action_plugins::action_plugin: lctl
Jan 22 20:25:10 mds1.local iml-agent-daemon[1668]: Jan 22 20:25:10.335  INFO iml_agent::action_plugins::action_plugin: ostpool_create
Jan 22 20:25:10 mds1.local iml-agent-daemon[1668]: Jan 22 20:25:10.335  INFO iml_agent::http_comms::session: Created new session PluginName("action_runner")/Id("8dbc5d3a-7e3e-4af5-b887-d423a1d155e4")
Jan 22 20:25:10 mds1.local iml-agent-daemon[1668]: Jan 22 20:25:10.459  INFO iml_agent::http_comms::session: Created new session PluginName("postoffice")/Id("e5d8b3b0-ddfa-48a3-ba3a-90671433eada")
Jan 22 20:25:10 mds1.local iml-agent-daemon[1668]: Jan 22 20:25:10.461  INFO iml_agent::http_comms::session: Created new session PluginName("stratagem")/Id("cdf5b76e-179a-431c-b3ba-dba75e0cf9af")
```

No errors on the postoffice:
```
[root@onyx-25 ~]# docker service logs -f iml_iml-postoffice
iml_iml-postoffice.1.ozj2zfd9k7gy@onyx-25.onyx.whamcloud.com    | Starting dependency check
iml_iml-postoffice.1.ozj2zfd9k7gy@onyx-25.onyx.whamcloud.com    | Waiting for dependencies
iml_iml-postoffice.1.ozj2zfd9k7gy@onyx-25.onyx.whamcloud.com    | Waiting for dependencies
iml_iml-postoffice.1.ozj2zfd9k7gy@onyx-25.onyx.whamcloud.com    | Dependency check passed

[root@onyx-25 ~]# docker stack ps iml | grep postoffice
ozj2zfd9k7gy        iml_iml-postoffice.1      imlteam/iml-postoffice:dev          onyx-25.onyx.whamcloud.com   Running             Running 2 hours ago
```

**Pull request merged**